src/GHC/Util/Unify.hs

```diff
@@ -109,6 +110,56 @@ unify' nm root x y
 unifyDef' :: Data a => NameMatch' -> a -> a -> Maybe (Subst' (LHsExpr GhcPs))
 unifyDef' nm x y = fmap mconcat . sequence =<< gzip (unify' nm False) x y
 
+unifyComposed' :: NameMatch'
+               -> LHsExpr GhcPs
+               -> LHsExpr GhcPs -> LHsExpr GhcPs -> LHsExpr GhcPs
+               -> Maybe (Subst' (LHsExpr GhcPs), Maybe (LHsExpr GhcPs))
+unifyComposed' nm x1 y11 dot y12 =
+  ((, Just y11) <$> unifyExp' nm False x1 y12)
+    <|> case y12 of
+          (L _ (OpApp _ y121 dot' y122)) | isDot dot' ->
+            unifyComposed' nm x1 (noLoc (OpApp noExt y11 dot y121)) dot' y122
+          _ -> Nothing
+
+-- unifyExp handles the cases where both x and y are HsApp, or y is OpApp. Otherwise,
+-- delegate to unifyExp'. These are the cases where we potentially need to call
+-- unifyComposed' to handle left composition.
+unifyExp :: NameMatch' -> Bool -> LHsExpr GhcPs -> LHsExpr GhcPs -> Maybe (Subst' (LHsExpr GhcPs), Maybe (LHsExpr GhcPs))
```

Comment:
Can you add a comment as to what the second component of the result is? When it is Just? If it is Just, what should be done with it?

Reply:
Done!